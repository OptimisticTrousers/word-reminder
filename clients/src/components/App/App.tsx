import { User } from "common";
import CSSModules from "react-css-modules";
import { Outlet } from "react-router-dom";

import styles from "./App.module.css";
import { Footer } from "../../layouts/Footer";
import { Navigation } from "../../layouts/Navigation";
import { EmailConfirmation } from "../../pages/Auth/EmailConfirmation";
import { useMobileTextSelectionAction } from "../../hooks/useMobileTextSelectionAction";

export const App = CSSModules(
  function ({ user }: { user: User }) {
    useMobileTextSelectionAction(String(user.id));

    if (import.meta.env.MODE === "production") {
      if (!user.confirmed) {
        return <EmailConfirmation user={user} />;
      }
    }

    return (
      <div styleName="app">
        <Navigation />
        <main styleName="app__main">
          <Outlet context={{ user }} />
        </main>
        <Footer />
      </div>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
