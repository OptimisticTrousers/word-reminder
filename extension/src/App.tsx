import { Outlet } from "react-router-dom";
import CSSModules from "react-css-modules";
import styles from "./App.module.css";

const App = CSSModules(() => {
  // const [colour, setColour] = useState("");
  // ("blue");
  // const onclick = async () => {
  //   let [tab] = await chrome.tabs.query({ active: true });
  //   chrome.scripting.executeScript<string[], void>({
  //     target: { tabId: tab.id! },
  //     args: [colour],
  //     func: (colour) => {
  //       document.body.style.backgroundColor = colour;
  //     },
  //   });
  // };

  return (
    <main styleName="main main--dark">
      <Outlet />
    </main>
  );
}, styles, { allowMultiple: true, handleNotFoundStyleName: "ignore" })

export default App;
