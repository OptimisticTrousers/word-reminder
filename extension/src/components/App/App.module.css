.app__main {
  position: relative;
  padding: 0.75rem;
}
