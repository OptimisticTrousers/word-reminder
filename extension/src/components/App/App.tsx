import { User } from "common";
import { useContext } from "react";
import CSSModules from "react-css-modules";
import { Outlet } from "react-router-dom";

import { ThemeContext } from "../../context/Theme";
import styles from "./App.module.css";
import { Footer } from "../../layouts/Footer";
import { Navigation } from "../../layouts/Navigation";
import { EmailConfirmation } from "../../pages/Auth/EmailConfirmation";

export const App = CSSModules(
  function ({ user }: { user: User }) {
    const { theme } = useContext(ThemeContext);

    if (!user.confirmed) {
      return <EmailConfirmation user={user} />;
    }

    return (
      <main styleName={`main main--${theme}`}>
        <Navigation />
        <Outlet context={{ user }} />
        <Footer />
      </main>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
