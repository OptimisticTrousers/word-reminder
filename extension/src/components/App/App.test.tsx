import { render, screen } from "@testing-library/react";

import { App } from "./App";

vi.mock("../../layouts/Footer", () => {
  return {
    Footer: function () {
      return <div data-testid="footer"></div>;
    },
  };
});

vi.mock("../../layouts/Navigation", () => {
  return {
    Navigation: function () {
      return <div data-testid="navigation"></div>;
    },
  };
});

vi.mock("react-router-dom", () => {
  return {
    Outlet: function () {
      return <div data-testid="outlet"></div>;
    },
  };
});

describe("App component", () => {
  it("renders main element and footer", () => {
    const { asFragment } = render(<App />);

    const footer = screen.getByTestId("footer");
    const navigation = screen.getByTestId("navigation");
    const outlet = screen.getByTestId("outlet");
    expect(footer).toBeInTheDocument();
    expect(navigation).toBeInTheDocument();
    expect(outlet).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
