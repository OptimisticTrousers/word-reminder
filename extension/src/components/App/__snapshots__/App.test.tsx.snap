// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders main element and footer 1`] = `
<DocumentFragment>
  <main
    class="_main_2f9351 _main--dark_2f9351"
  >
    <div
      data-testid="navigation"
    />
    <div
      data-testid="outlet"
    />
    <div
      data-testid="footer"
    />
  </main>
</DocumentFragment>
`;
